bin/app.ts

```diff
@@ -159,16 +160,54 @@ export class RoutingAPIPipeline extends Stack {
       secretCompleteArn: 'arn:aws:secretsmanager:us-east-2:644039819003:secret:routing-api-internal-api-key-Z68NmB',
     })
 
+    // Read chains that uses RPC gateway.
+    const rpcGatewayEnabledChainIds = getRpcGatewayEnabledChainIds()
+
     // Load RPC provider URLs from AWS secret
     let jsonRpcProviders = {} as { [chainId: string]: string }
     SUPPORTED_CHAINS.forEach((chainId: ChainId) => {
-      const key = `WEB3_RPC_${chainId}`
-      jsonRpcProviders[key] = jsonRpcProvidersSecret.secretValueFromJson(key).toString()
-      new CfnOutput(this, key, {
-        value: jsonRpcProviders[key],
-      })
+      if (!rpcGatewayEnabledChainIds.includes(chainId)) {
+        const key = `WEB3_RPC_${chainId}`
+        jsonRpcProviders[key] = jsonRpcProvidersSecret.secretValueFromJson(key).toString()
+        new CfnOutput(this, key, {
+          value: jsonRpcProviders[key],
+        })
+      }
     })
 
+    // Load RPC provider URLs from AWS secret (for RPC Gateway)
+    const RPC_GATEWAY_PROVIDERS = [
+      // Avalanche
+      'INFURA_43114',
+      'NIRVANA_43114',
+      'QUICKNODE_43114',
+      // Optimism
+      'INFURA_10',
+      'NIRVANA_10',
+      'QUICKNODE_10',
+      'ALCHEMY_10',
+      // Celo
+      'INFURA_42220',
+      'QUICKNODE_42220',
+      // BNB
+      'QUICKNODE_56',
+      // Polygon
+      'INFURA_137',
+      'QUICKNODE_137',
+      'ALCHEMY_137',
+      // Base
+      'INFURA_8453',
+      'QUICKNODE_8453',
+      'ALCHEMY_8453',
+      'NIRVANA_8453',
+    ]
+    for (const provider of RPC_GATEWAY_PROVIDERS) {
+      jsonRpcProviders[provider] = jsonRpcProvidersSecret.secretValueFromJson(provider).toString()
```

Comment:
I think this could take us over the 4kb limit.

Is there any way we could simplify the secrets? Like all we need for Infura is an API key, every URL can be hardcoded based on the chain, I think there are some providers that return random urls, but it would be helpful to figure out if there's a way to simplify this, we are essentially multiplying our providers for every chain which is not gonna work as we need to expand to new chains.

Reply:
I had the same suggestion https://uniswapteam.slack.com/archives/C021SU4PMR7/p1710359540530009?thread_ts=1710353824.086039&cid=C021SU4PMR7

Reply:
Especially with rollup-centric road and rollup-as-a-service offering, we will likely roll out to new chain at faster speed.

Reply:
@mikeki Made some changes to reduce the env var's length. Please take a look!